servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/LoadBalancedStreamingHttpClient.java

```diff
@@ -69,6 +73,11 @@
         // following the LoadBalancer API which this Client depends upon to ensure the concurrent request count state is
         // correct.
         return loadBalancer.selectConnection(SELECTOR_FOR_REQUEST).flatMap(c -> {
+                final Consumer<ConnectionInfo> onConnectionSelected = AsyncContext.get(ON_CONNECTION_SELECTED_CONSUMER);
+                if (onConnectionSelected != null) {
+                    onConnectionSelected.accept(c.connectionContext());
+                    AsyncContext.remove(ON_CONNECTION_SELECTED_CONSUMER);
```

Comment:
We don't need to worry about race, because it's not expected within a reactive streams chain of operators. All events are sequential. 

Yes, if the `accept` throws, it's preferable to keep the consumer to let the retry filter (which can be applied after the observer filter) recover. However, we don't need to remove the consumer from the async context in any case, because something else may fail, and retry will have to select a new connection. Thanks for pointing to this!

Reply:
Fixed in https://github.com/apple/servicetalk/pull/1824/commits/e815e0e38afbf992ac70b188a9becc9de965da40